### ISSUE TITLE
Question 1: Data augmentation

### ISSUE THREAD

**Issue by AboHadi313** [ndrplz/self-driving-car#4]
Hello sir,
Here shouldn't be `X_train` instead of `X_train_norm` inside ` train_test_split()`?
`X_train_norm, X_val_norm, y_train, y_val = train_test_split(X_train_norm, y_train, test_size=VAL_RATIO, random_state=0)`
Anyway, here `img_rgb = X_train[0]`, you used X_train[0] which will always give me the first picture of the training dataset! will never be random.
And here: `plt.title('Example of RGB image (class = {})'.format(y_train[0]))`, you used the new training labels (after splitation).

Not sure if you got my point, but it is not correct, I am new to python so I couldn't figure out how to fix the issue here, so any help?

**Comment by ndrplz** [comment]
Hi @AboHadi313 ,
to which file are you referring to?

**Comment by AboHadi313** [comment]
Hello,
[Here](https://github.com/ndrplz/self-driving-car/blob/master/project_2_traffic_sign_classifier/Traffic_Sign_Classifier.ipynb)

```
from sklearn.model_selection import train_test_split
from keras.preprocessing.image import ImageDataGenerator

# split into train and validation
VAL_RATIO = 0.2
X_train_norm, X_val_norm, y_train, y_val = train_test_split(X_train_norm, y_train, test_size=VAL_RATIO, random_state=0)


# create the generator to perform online data augmentation
image_datagen = ImageDataGenerator(rotation_range=15.,
                                   zoom_range=0.2,
                                   width_shift_range=0.1,
                                   height_shift_range=0.1)

# take a random image from the training set
img_rgb = X_train[0]

# plot the original image
plt.figure(figsize=(1,1))
plt.imshow(img_rgb)
plt.title('Example of RGB image (class = {})'.format(y_train[0]))
plt.show()

# plot some randomly augmented images
rows, cols = 4, 10
fig, ax_array = plt.subplots(rows, cols)
for ax in ax_array.ravel():
    augmented_img, _ = image_datagen.flow(np.expand_dims(img_rgb, 0), y_train[0:1]).next()
    ax.imshow(np.uint8(np.squeeze(augmented_img)))
plt.setp([a.get_xticklabels() for a in ax_array.ravel()], visible=False)
plt.setp([a.get_yticklabels() for a in ax_array.ravel()], visible=False)
plt.suptitle('Random examples of data augmentation (starting from the previous image)')
plt.show()
```

**Comment by ndrplz** [comment]
Nice catch! it's a bug in the visualization.

The fact is that after preprocessing with
`X_train_norm = preprocess_features(X_train)
`
the images are ugly to visualize (since they have been standardized in small values around zero).

So I wanted to show an image in the plot before preprocessing. However, I forgot that due to the shuffling the images no longer correspond to the labels. 

For the purpose of training the classifier, this has no effect. However, if you want to visualize the right image you can substitute  `X_train[0]` to  `X_train_norm[0]` as you noticed.
